storage/innobase/handler/ha_innodb.cc

```diff
@@ -3691,6 +3691,23 @@ innobase_init(
 	srv_kill_idle_transaction = 0;
 #endif
 
+
+	/* Do not enable backoff algorithm for small buffer pool.
+	Check innodb_srv_empty_free_list_algorithm_validate
+	for additional information */
+	if ( innobase_buffer_pool_size / srv_page_size
```

Comment:
This condition should be extracted to its own function, and used from both here and further in ..._validate, and the big explaining comment moved to it.